copier.go

```diff
@@ -66,5 +69,91 @@ func TestCopier(t *testing.T, store types.Storager) {
 				})
 			})
 		})
+
+		Convey("When Copy to an existing file", func() {
+			srcSize := rand.Int63n(4 * 1024 * 1024) // Max file size is 4MB
+			content, _ := ioutil.ReadAll(io.LimitReader(randbytes.NewRand(), srcSize))
+			src := uuid.New().String()
+
+			_, err := store.Write(src, bytes.NewReader(content), srcSize)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			dstSize := rand.Int63n(4 * 1024 * 1024) // Max file size is 4MB
+			r := io.LimitReader(randbytes.NewRand(), dstSize)
+			dst := uuid.New().String()
+
+			_, err = store.Write(dst, r, dstSize)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			defer func() {
+				err = store.Delete(src)
+				if err != nil {
+					t.Error(err)
+				}
```

Comment:
This part should be moved ahead, otherwise `src` won't be deleted when `dst` has an error.